### ISSUE TITLE
Default values for Input Objects do not work with Enum types

### ISSUE THREAD

**Issue by nickblumenthal** [rmosolgo/graphql-ruby#1826]
GraphQL v1.8.8

When you have an enum that has a different underlying value than its name, using that value as a default value for an input field in a mutation does not work.

Here's the example:

We have an enum that looks like this: 

```
AgentStateEnum = GraphQL::EnumType.define do
  name 'AgentStateEnum'
  description 'State of an Agent\'s account.'

  value 'ACTIVE', 'Represents an authorized agent in our system.', value: 'active'
  value 'DISABLED', 'Agent is disabled, web app access is denied.', value: 'disabled'
  value 'ROSTER_AGENT', 'Agent exists in the system, awaiting activation.', value: 'roster_agent'
end
```

and a mutation like this:

```
CreateAgentMutation = GraphQL::Relay::Mutation.define do
  name 'CreateAgent'
  description 'Create an agent. Automatically associated with the currently authenticated access token owner.'

  # Input fields
  input_field :email, !types.String,
    'Email address of Agent. Update with care as this also acts as the agent\'s login. Must be a properly formatted email address.'
  input_field :state, AgentStateEnum, 'State of the Agent\'s account. Defaults to "active".', default_value: 'active'

  # Return fields
  return_field :agent, AgentType
  resolve CreateAgentResolver.new
end
```

However, if we execute the mutation without a variable for `state`, that input is `nil` in the resolver (instead of `active` like you would expect).

This is due to https://github.com/rmosolgo/graphql-ruby/commit/4eaeb98472fab50dca321678cd4bd3a85b4e3472

That commit attempts to coerce the default value via the input field's type. Unfortunately, an Enum types coercion is done by name, not by value.  

I thought maybe I could just set the default_value to the name instead like so:
```
 input_field :state, AgentStateEnum, 'State of the Agent\'s account. Defaults to "active".', default_value: 'ACTIVE'
```

Unfortunately this does not pass validation because the validator ultimately calls `coerce_result` which compares the default value to the enumerator's underlying ruby value (https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/enum_type.rb#L113)

While a simple solution would be to change https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/input_object_type.rb#L96 to 
```
coerced_value = input_field_defn.type.coerce_input(input_field_defn.default_value, ctx) || input_field_defn.default_value
```

,it seems as if the root of the problem is whether to set default values on input fields of an enum type by name or by value.  I'm wondering if perhaps I'm just using input fields incorrectly, if the solution needs to be refactoring the `EnumType`, or if the above solution would suffice for a PR.

Thanks!

**Comment by nickblumenthal** [comment]
See this PR for my idea https://github.com/rmosolgo/graphql-ruby/pull/1827

**Comment by rmosolgo** [comment]
Fixed by #1827
